aopy/utils.py

```diff
@@ -429,4 +431,54 @@ def calc_radial_dist(pos, origin=(0,0)):
     for i, p in enumerate(pos):
         dist[i] = math.dist(origin, p)
     
-    return dist
\ No newline at end of file
+    return dist
+
+'''
+Memory
+'''
+def get_memory_available_gb():
+    '''
+    Get the available system memory in gigabytes. Only works on linux platforms.
+
+    Returns:
+        int: number of gigabytes of available system memory
+    '''
+    if platform.system() != "Linux":
+        print('Only works on linux!')
+        return
+    with open('/proc/meminfo', 'r') as mem:
+        free_memory = 0
+        for i in mem:
+            sline = i.split()
+            if str(sline[0]) == 'MemAvailable:':
+                free_memory = int(sline[1])
+                break
+    return int(free_memory/1e6) # convert KB to GB
+
+def set_memory_limit_gb(size_gb):
+    '''
+    Set a memory resource limit in gigabytes.
+
+    Args:
+        size_gb (int): upper limit of memory that will be made available to python in gigabytes
+    '''
+    maxsize = int(1e9*size_gb)
+    resource.setrlimit(resource.RLIMIT_AS, (maxsize, -1))
+
+def get_memory_limit_gb():
+    '''
+    Get the memory resource limit in gigabytes.
+
+    Returns:
+        int or None: upper limit of memory available to python in gigabytes
+    '''
+    soft, hard = resource.getrlimit(resource.RLIMIT_AS)
```

Comment:
Adding some more information about the soft and hard limits. The soft limit is a value upon which the operating system will notify the process via a signal or typically restrict it. An upper bound on the memory values is defined by the hard limit. Although the hard limit can be lowered, it can never be raised by user processes (even if the process lowered itself) and is controlled by a system-wide parameter set by the system administrator.